src/map.rs

```diff
@@ -178,7 +180,6 @@ where
     tonic::transport::Server::builder()
         .add_service(map_server::MapServer::new(map_svc))
         .serve_with_incoming(_uds_stream)
-        .await?;
-
-    Ok(())
+        .await
+        .map_err(Into::into)
```

Comment:
```suggestion
        .await?
```

just curious, won't `?` do an implicit `Into::into`?

Reply:
Yes. Since this is the last statement, I removed `?`. 
So it can be either:
```rust
.await?; 
Ok(()) // final return Ok variant
```
Or just end with that last await call and return its result directly.